src/roving-tabindex/use-roving-tabindex.js

```diff
@@ -0,0 +1,105 @@
+import { useRef, useContext, useLayoutEffect, useCallback } from 'react'
+import { RovingTabIndexContext, ActionTypes } from './tabindex-provider'
+
+let idCounter = 0
+
+const uniqueId = (prefix) => {
+  const id = ++idCounter
+  return `prefix_${id}`
+}
+
+const TabDirection = {
+  Next: 'Next',
+  Previous: 'Previous'
+}
+
+export default function useRovingTabIndex({
+  disabled,
+  isSelectable,
+  isSelected,
+  ref = null
+}) {
+  const tabIndexId = useRef(uniqueId('eg-roving-tabindex_'))
```

Comment:
Why not https://github.com/segmentio/evergreen/blob/master/src/hooks/use-id.js

Reply:
Forgot we had that, that's a great improvment